Sources/BottomSheetCalculator.swift

```diff
@@ -30,68 +30,98 @@ struct BottomSheetCalculator {
         return max(superview.frame.height - makeTargetHeight(), handleHeight)
     }
 
-    /// Calculates bottom and top thresholds for the given target offsets.
+    /// Creates the layout of the BottomSheetView based on the target offsets and the current target offset
+    ///
     /// - Parameters:
-    ///   - contentView: the content view of the bottom sheet
-    ///   - superview: the bottom sheet container view
-    ///   - height: preferred height for the content view
-    static func thresholds(for targetOffsets: [CGFloat], in superview: UIView) -> [CGFloat] {
+    ///   - targetOffsets: array containing the different target offsets a BottomSheetView can transition between
+    ///   - currentTargetIndex: index of the current target offset of the BottomSheetView
+    ///   - isDismissable: flag specifying whether the last two offsets should dismiss the BottomSheetView
+    static func createLayout(for targetOffsets: [CGFloat], at currentTargetIndex: Int, isDismissible: Bool) -> [BottomSheetModel] {
         guard !targetOffsets.isEmpty else { return [] }
 
+        let minOffset = targetOffsets.last ?? 0
+        let maxOffset = targetOffsets.first ?? 0
         let maxThreshold: CGFloat = 75
-        let targetOffsets = [0] + targetOffsets + [superview.frame.height]
 
-        return zip(targetOffsets.dropFirst(), targetOffsets).map {
-            min(abs(($0 - $1) * 0.25), maxThreshold)
+        // [0, thresholds, 0]
+        // Add 0s to array to limit translation above and below edge offsets
+        let thresholds = [0] + targetOffsets.mapPar { (first, second) -> CGFloat in
```

Comment:
Is this `mapPar` the same as `zip(targetOffsets.dropFirst(), targetOffsets).map`?